### DISCUSSION TITLE
Welcome to scratch-status Discussions!

### DISCUSSION THREAD

**Discussion by toakiryu** [toakiryu/scratch-status#5] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by toakiryu** [comment]
# Scratch Status 開発移行のお知らせ

>  [!NOTE]
> 「危険なサイト」という表示についてご説明します。
> `toakiryu.com` 親ドメインを利用して公開している一部コンテンツで[ソーシャルエンジニアリングコンテンツ](https://developers.google.com/search/docs/monitor-debug/security/social-engineering)が検出されました。そのコンテンツと同じ親ドメインを利用しているすべてのコンテンツに警告が表示されています。この問題は以前から度々確認されています。
> 
> 問題となったコンテンツは「[Next.js Rich Template](https://github.com/toaki-cltv/nextjs-rich-tpl)」です。このコンテンツは [Next.js のデフォルトテンプレート](https://vercel.com/templates/next.js/nextjs-boilerplate)を元として追加で様々な機能や改良をしてアップグレードしたテンプレートです。テンプレートのため大きなデザイン変更を行っておらず、それがフィッシングコンテンツとして検出されたと解釈しています。

本プロジェクト「Scratch Status」は、今後の主要な開発を終了し、  
後継プロジェクト **[Scratch Status Monitor](https://github.com/scratchcore/scratch-status-monitor)** に開発の主軸を移行しました。

なお、本リポジトリは今後も存続し、  
**致命的な不具合やセキュリティ上の問題が確認された場合には、必要に応じて修正対応を行う予定です。**

<details>
<summary>詳細はこちら</summary>

本サービスは当初、個人利用を目的として開発されたものであり、  
大規模なアクセスや長期運用を前提とした設計ではありませんでした。

リリース直後、[Vercel ](https://vercel.com/home)の無料枠制限により運営が難しくなったため、  
一時的にサイトを閉鎖し、その後 [AWS Amplify](https://aws.amazon.com/jp/amplify/) へ移行して 12 ヶ月間の無料枠で運営を継続してきました。

しかし、現行構成では最適化が不十分で、  
運用コストやリソース効率の面で課題が残っていました。  
そのため、アーキテクチャを基礎から見直し、より効率的で拡張性の高い構成を目指す形で  
後継プロジェクト **[Scratch Status Monitor](https://github.com/scratchcore/scratch-status-monitor)** の開発を進めています。

新プロジェクトでは、  
運用性・パフォーマンス・メンテナンス性を大幅に改善し、  
より安定した監視環境の提供を目指しています。

なお、後継プロジェクトが一般公開されるまでの間は、  
本サービスを継続して一般公開する予定です。

> 新プロジェクトの開発中の画像
<img width="318" height="854" alt="image" src="https://github.com/user-attachments/assets/7cd8f62b-9d32-43d5-844d-e3a48cd4709b" />
<img width="1913" height="498" alt="image" src="https://github.com/user-attachments/assets/e91579d2-94d8-4eb7-b483-62abebe657c6" />

</details>

---

## 変更アーカイブ
<details>
<summary>2025/2/18: サイトのアクセス制限の可能性...</summary>

### サイトのアクセス制限の可能性について

いつも当サイトをご利用いただき、誠にありがとうございます。

現在、当サイトには多くのユーザーにご利用いただいており、誠に感謝しております。しかしながら、現在の利用状況において、リクエストの上限に近づいています。具体的には、Vercelの「Function Invocations」の使用量が既に78,613回となり、残りのリソースは100,000回までとなっています。このままでは、サービスの提供が継続困難となる可能性があります。

今後もサイトを安定的に運営するためには、追加の支援が必要です。もし支援をいただける場合、サイトの継続的な運営が可能になります。それが難しい場合、サービスを一時的に閉鎖またはアクセスを制限せざるを得ない状況となります。

ユーザーの皆さまにはご不便をおかけすることになり、大変申し訳ございません。ご理解とご協力をいただけますよう、お願い申し上げます。

今後とも、どうぞよろしくお願い申し上げます。
</details>

**Comment by syun-scratch** [comment]
こんにちは。
「不明なエラーが発生」と表示されてしまいました。
再読み込み等しましたが直りそうではありません。
どうすれば良いでしょうか。

**Reply by QWE-sava** [reply]
scratch自体のサーバーが現在バグっていると思われるのでエラーが出ているのだと思われます。

**Reply by toakiryu** [reply]
このサービスは Scratch関連サーバーのステータスを取得して、その情報を見やすく表示しているだけです。もし取得に失敗したらその原因が表示されるようになっており、現在を原因を探しています。
サーバー側に問題が発生した場合、専用のページにリダイレクトされるので、「不明なエラーが発生」と表示された場合、おそらくクライアント側で問題が発生した可能性が高いです。
もしエラーに遭遇したら、ブラウザのディベロッパーツールを開き、ページとログをスクリーンショット等で記録していただけると大変助かります。

**Reply by syun-scratch** [reply]
丁寧なご回答ありがとうございます。
もし次遭遇した場合は出来る限りそのようにします。

**Comment by YU-TechnicalDepartment** [comment]
こんにちは。Scratchでの、YU1000です。
バグの報告なのですが、コメントが読み込まれてから言語を変更すると、コメントがページの上部に移動してしまうバグを発見しました。
急いで直して欲しいわけではないのですが、念のため報告させていただきました。
よろしくお願いします。

**Reply by toakiryu** [reply]
ご報告ありがとうございます。

当サイトは [giscus](https://giscus.app/ja) という外部システムを利用してコメントをサイトに埋め込んでおり、この問題は以前から確認されています。

### 原因
言語変更を行うとページの要素を再レンダリングします。しかし `giscus` の埋め込み要素は再レンダリングされず、それ以外の要素は再レンダリングされる為、要素の順番が変わって表示されています。

### 対処
この問題への対処は現在検討していません。
現在後継サービスの開発を行っており、致命的な問題に限り対応を行う方針です。

後継サービス：[Scratch Status Monitor](https://github.com/scratchcore/scratch-status-monitor)
